File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int el = nums[0];
        int c = 1;
        for (int i = 1; i < nums.size(); i++){
            if (el == nums[i]){
                c++;
            }
            else{
                c--;
            }
            if (c==0){
                el = nums[i];
                c = 1;
            }
        }

        return el;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}
        count = [[] for i in range(len(nums) + 1)]

        for i in nums:
            freq[i] = freq.get(i, 0) + 1
        
        for key, v in freq.items():
            count[v].append(key)

        res = []

        for i in range(len(count) - 1, 0, -1):
            for n in count[i]:
                res.append(n)
                if len(res) >= k:
                    return res